windows/src/engine/kmcomapi/processes/keyboard/Keyman.System.Process.KPInstallKeyboardLanguage.pas

```diff
@@ -208,14 +209,28 @@ function TKPInstallKeyboardLanguage.ConvertBCP47TagToLangID(Locale: string;
   end
   else
   begin
-    // Assuming that the tag is a Language-Script-Region triplet at most.
-    // If you use -Variant or -Extension then YMMV.
-    with TBCP47Tag.Create(Locale) do
+    //If tag is a Language-Script-Region triplet at most, then we can use
+    // LocaleNameTOLCID on the Language-Region to get a LangID. If you use
+    // -Variant or -Extension then we won't get a valid LangID. On Win7/8,
+    // this will result in installation under default locale.
+
+    tag1 := TBCP47Tag.Create(Locale);
     try
-      Script := '';
-      Locale := Tag;
+      tag1.Script := '';
+      tag2 := TBCP47Tag.Create(tag1.Language);
+      try
+        tag2.Region := tag1.Region;
+        if tag2.Tag <> tag1.Tag then
+        begin
+          // We have a complex tag, Windows won't convert it with LocaleNameToLCID
+          Exit(False);
```

Comment:
Does LangID stay the same?

Reply:
In this situation, the langid will be a transient language because it's a complex tag such as `en-US-fonipa`. So we return `False`.